src/Adapter/Module/Repository/ModuleRepository.php

```diff
@@ -94,17 +101,78 @@ public function getActiveModulesPaths(): array
     {
         if (null === $this->activeModulesPaths) {
             $this->activeModulesPaths = [];
-            $modulesFiles = Finder::create()->directories()->in(_PS_MODULE_DIR_)->depth(0);
             $activeModules = $this->getActiveModules();
 
-            foreach ($modulesFiles as $moduleFile) {
-                $moduleName = $moduleFile->getFilename();
+            foreach ($this->getModules() as $moduleName => $modulePath) {
                 if (in_array($moduleName, $activeModules)) {
-                    $this->activeModulesPaths[$moduleName] = $moduleFile->getPathname();
+                    $this->activeModulesPaths[$moduleName] = $modulePath;
                 }
             }
         }
 
         return $this->activeModulesPaths;
     }
+
+    /**
+     * Returns an array of native modules
+     *
+     * @return array<string>
+     */
+    public function getNativeModules(): array
+    {
+        // Native modules are the one integrated in PrestaShop release via composer
+        // so we use the lock files to generate the list
+        $content = file_get_contents(_PS_ROOT_DIR_ . '/composer.lock');
```

Comment:
This file path should be a parameter from the constructor This will allow you to really unit test the behavior with a dedicated resource composer.lock file and check all the detected moduels

Reply:
Fixed